core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -270,4 +270,261 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
     generatedColumnsAndColumnsUsedByGeneratedColumns.map(_.toLowerCase(Locale.ROOT)).toSet
   }
 
+  /**
+   * Try to get `OptimizablePartitionExpression`s of a data column when a partition column is
+   * defined as a generated column and refers to this data column.
+   *
+   * @param schema the table schema
+   * @param partitionSchema the partition schema. If a partition column is defined as a generated
+   *                        column, its column metadata should contain the generation expression.
+   */
+  def getOptimizablePartitionExpressions(
+      schema: StructType,
+      partitionSchema: StructType): Map[String, Seq[OptimizablePartitionExpression]] = {
+    val partitionGenerationExprs = partitionSchema.flatMap { col =>
+      getGenerationExpressionStr(col).map { exprStr =>
+        val expr = parseGenerationExpression(SparkSession.active, exprStr)
+        new Column(expr).alias(col.name)
+      }
+    }
+    if (partitionGenerationExprs.isEmpty) {
+      return Map.empty
+    }
+
+    val spark = SparkSession.active
```

Comment:
is this the right way to get spark session? cant the session be explicitly passed to this method?